tests/unittest_qmail-send.c

```diff
@@ -0,0 +1,254 @@
+#include <stdio.h>
+#define DEPRECATED_FUNCTIONS_REMOVED
+#include <check.h>
+#include "qmail-send-without-main.c"
+
+extern struct job *jo;
+extern int numjobs;
+
+// START test helpers for the API we might prefer from qmail-send
+
+#define available 0
+#define unavailable 1
+
+static void job_init_this_many(int numslots) 
+{
+  numjobs = numslots;
+  job_init();
+}
+
+static void occupy_job(unsigned int index) 
+{
+  jo[index].refs = unavailable;
+}
+
+static void occupy_all_jobs(void) 
+{
+  int j;
+  for (j = 0; j < numjobs; j++) 
+  {
+    occupy_job(j);
+  }
+}
+
+static void release_job(unsigned int index) 
+{
+  jo[index].refs = available;
+}
+
+// END test helpers for the API we might prefer from qmail-send
+
+// when job_avail() returns an index, switch to ck_assert_int_eq
+#define assert_job_avail ck_assert_uint_eq
+
+static void assert_job_at_index(unsigned int index, int availability) 
+{
+  ck_assert_int_eq(jo[index].refs, availability);
+}
+
+static void setup(void) 
+{
+  numjobs = -77;
+}
+
+static void teardown(void) 
+{
+  numjobs = -88;
+}
+
+static void assert_all_slots_are_available(int numjobs) 
+{
+  int j;
+  for (j = 0; j < numjobs; j++) 
+  {
+    assert_job_at_index(j, available);
+    ck_assert_ptr_eq(jo[j].sender.s, 0);
+  }
+}
+
+START_TEST(test_qmail_job_init_sets_expected_default_state)
+{
+  ck_assert_ptr_null(jo);
+
+  job_init_this_many(4);
+
+  ck_assert_ptr_nonnull(jo);
+  ck_assert_uint_eq(numjobs, 4);
+  assert_all_slots_are_available(4);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_free_slots_after_init)
+{
+  job_init_this_many(5);
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_no_free_slots)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+
+  assert_job_avail(job_avail(), 0);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(2);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot_for_index_zero)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(0);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_avail_returns_a_free_slot_for_last_index)
+{
+  job_init_this_many(5);
+  occupy_all_jobs();
+  release_job(numjobs - 1);
+
+  assert_job_avail(job_avail(), 1);
+}
+END_TEST
+
+START_TEST(test_qmail_job_open_slots_full)
+{
+  int index;	
+  job_init_this_many(2);
```

Comment:
Please add a newline after the variable just for consistency and easier reading.